torch/nn/functional.py

```diff
@@ -3818,6 +3824,16 @@ def interpolate(input: Tensor, size: Optional[int] = None, scale_factor: Optiona
     if input.dim() == 5 and mode == "nearest":
         return torch._C._nn.upsample_nearest3d(input, output_size, scale_factors)
 
+    # TODO: Remove this scripting logic once the 2-week FC window has passed.
+    if mode == "nearest-exact" and torch.jit.is_scripting():
+        raise RuntimeError("TorchScript currently does not support nearest-exact")
+    if input.dim() == 3 and mode == "nearest-exact":
+        return torch._C._nn._upsample_nearest_exact1d(input, output_size, scale_factors)
+    if input.dim() == 4 and mode == "nearest-exact":
+        return torch._C._nn._upsample_nearest_exact2d(input, output_size, scale_factors)
+    if input.dim() == 5 and mode == "nearest-exact":
+        return torch._C._nn._upsample_nearest_exact3d(input, output_size, scale_factors)
+
```

Comment:
I mentioned this on the other interpolate PR, but the `torch.jit.is_scripting()` gating needs to be done slightly differently. In particular, it should not be possible to enter a branch when scripting is enabled that contains a call to a new op. Scripting serializes all possible branches rather than executing linearly. One possibility:

```python
if mode == "nearest-exact":
    if not torch.jit.is_scripting():
        if input.dim() == 3 and mode == "nearest-exact":
            return torch._C._nn._upsample_nearest_exact1d(input, output_size, scale_factors)
        if input.dim() == 4 and mode == "nearest-exact":
            return torch._C._nn._upsample_nearest_exact2d(input, output_size, scale_factors)
        if input.dim() == 5 and mode == "nearest-exact":
            return torch._C._nn._upsample_nearest_exact3d(input, output_size, scale_factors)
    else:
        raise RuntimeError("TorchScript currently does not support nearest-exact")
```

Reply:
Thanks for the suggestion. I applied that in the latest commit. Could you please check that. Thanks!